sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```
@@ -370,6 +371,10 @@ abstract class JdbcDialect extends Serializable with Logging{
    * returns whether the dialect supports limit or not
    */
   def supportsLimit(): Boolean = true
+
+  def supportsTableSample: Boolean = false
+
+  def getTableSample(sample: TableSampleInfo): String = ""
```

shall we throw `UnsupportedOperationException` by default?